x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingContext.java

```diff
@@ -180,6 +180,13 @@ public boolean isEmpty() {
             && modelAssignments.isEmpty();
     }
 
+    public boolean hasOnlyZeroAllocationModels() {
```

Comment:
Maybe this this should replace the `isEmpty()` function above 


```
        return anomalyDetectionTasks.isEmpty()
            && snapshotUpgradeTasks.isEmpty()
            && dataframeAnalyticsTasks.isEmpty()
            && (modelAssignments.isEmpty() || modelAssignments.values().stream().allMatch(assignment -> assignment.totalTargetAllocations() == 0));
```

MlAutoscalingContext.isEmpty() is also called in the assertion in `MlMemoryAutoscalingDecider` line 232 so it would be good to have to same logic there

Reply:
I agree; fixed